src/mesh.jl

```diff
@@ -1387,7 +1384,7 @@ function fill_face_boundary_mesh_topology!(topo,mesh,D,d)
                     dfaces = vertex_to_dfaces[vertex]
                     for dface1 in dfaces
                         vertices1 = dface_to_vertices[dface1]
-                        if same_valid_ids(vertices,vertices1)
+                        if same_valid_ids(collect(vertices),collect(vertices1))
```

Comment:
By adding collect, you are allocating a new array (we want to avoid this). The same is also true for other instances below.

Reply:
Thanks; very useful info: I removed recollect but now the vector types must be `AbstractVector`s I suppose.